### PR TITLE
Add some uniqueness checks to models

### PR DESCRIPTION
**Pull request by cg2v**
Several models should have uniqueness constraints. Missing uniqueness
constraints can trigger mysql errors when the database has unique keys
for some columns.

### CODE REVIEW

File: app/models/problem.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Problem < ActiveRecord::Base
   belongs_to :assessment, touch: true
   has_many :annotations
 
-  validates :name, presence: true
+  validates :name, presence: true, uniqueness: {scope: :assessment_id}
```
Comment (dlbucci) [inline app/models/problem.rb]: Can you check these files with rubocop?  I think validates with a :uniqueness argument is the preferred way to add this check.  If so, I think the other files should be edited to use that method instead, just to keep the code slightly more up to date with conventions.

**Comment by cg2v** [comment]
The only thing it didn't like was the (lack of) spaces on the line you quoted

**Comment by dlbucci** [comment]
Agreed, it looks better with the spaces :)

**Comment by nayak16** [comment]
Develop branch deleted. Will add uniqueness checks in new branch

**Pull request closed without merging**